File: 205. Isomorphic_Strings/solution.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        
        int check = 1;
        Map<Character,Character> dict = new HashMap<Character,Character>();
        Map<Character,Character> dict2 = new HashMap<Character,Character>();
        
        for(int i = 0 ; i < s.length() ; i++){
            if(!dict.containsKey(s.charAt(i)))
                dict.put(s.charAt(i), t.charAt(i));
            else if(dict.get(s.charAt(i)) != t.charAt(i)){
                check = 0;
                break;
            }
            
            if(!dict2.containsKey(t.charAt(i)))
                dict2.put(t.charAt(i),s.charAt(i));
            else if(dict2.get(t.charAt(i)) != s.charAt(i)){
                check = 0;
                break;
            }
            
        }

        return check == 1 ? true : false;
        
        
    }
}

File: 205. Isomorphic_Strings/solution.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {

        int check = 1;
        unordered_map<int,int> dict;
        unordered_map<int,int> dict2;
        
        for(int i = 0 ; i < s.length() ; i++){
            if(dict.find(s[i]) == dict.end())
                dict[s[i]] = t[i];
            else if(dict[s[i]] != t[i]){
                check = 0;
                break;
            }
            
            if(dict2.find(t[i]) == dict2.end())
                dict2[t[i]] = s[i];
            else if(dict2[t[i]] != s[i]){
                check = 0;
                break;
            }
            
        }

        return check == 1 ? true : false;
        
    }
};